code/workspaces/web-app/src/config/catalogue.js

```diff
@@ -1,11 +1,15 @@
 import axios from 'axios';
 
-export async function catalogueConfig() {
-  const { data: { catalogue } } = await axios.get('/catalogue_asset_repo_config.json');
-  return catalogue;
-}
+// This is a cached value.
+// Initialise asynchronously before React starts.
+// Get synchronously within React.
+let catalogue;
 
-export async function catalogueAvailable() {
-  const config = await catalogueConfig();
-  return config.available;
+async function initialiseCatalogue() {
+  const { data } = await axios.get('/catalogue_asset_repo_config.json');
+  ({ catalogue } = data);
```

Comment:
Are the surrounding `()` needed? Also did not know that you could destructure into an existing variable.

Reply:
As discussed, it looks like the surrounding brackets are what let you destructure into an existing variable, see https://stackoverflow.com/questions/48714689/javascript-re-assign-let-variable-with-destructuring